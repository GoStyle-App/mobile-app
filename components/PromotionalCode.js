import React, { useState } from 'react';
import { StyleSheet, View, Text, } from 'react-native';
import { IconButton, Colors } from 'react-native-paper';

export default function PromotionalCode(codes) {
    const code = codes.code;
    const navigation = codes.navigation;
    const myCodes = codes.myCodes;
    const [ hidden, setHidden ] = useState(false);

    if (codes.detail === false) {
        return (
            <View style={ styles.list }>
                <Text style={ styles.code }>
                    { code.code }
                </Text>
                <Text style={ styles.label }>
                    { code.label }
                </Text>
            </View>
        );
    }

    if (hidden) {
        return (
            <View/>
        );
    }

    return (
        <View style={ styles.list }>
            <Text style={ styles.code }>
                { code.code }
            </Text>
            <Text style={ styles.label }>
                { code.label }
            </Text>
            <IconButton
                icon="information"
                color={ Colors.blue300 }
                size={ 20 }
                onPress={() => {
                    navigation.navigate('DetailPromotionalCode', {
                        code: code,
                    });
                }}
            />
            <IconButton
                icon="trash-can-outline"
                color={ Colors.red500 }
                size={ 20 }
                onPress={() => {
                    const index = myCodes.findIndex(x => x.id === code.id);
                    if (index !== undefined) {
                        myCodes.splice(index, 1);
                    }
                    setHidden(true);
                }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    list: {
        flexDirection: 'row',
        textAlign: 'left',
        alignItems: 'center',
        marginBottom: 10,
    },
    code: {
        width: 80,
        backgroundColor: '#000',
        padding: 2,
        color: '#ffffff',
        textAlign: 'center',
    },
    label: {
        width: 150,
        marginLeft: 10,
        textAlign: 'left',
    }
});
