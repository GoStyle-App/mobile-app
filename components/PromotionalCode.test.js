import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import PromotionalCode from './PromotionalCode';

describe('Testing PromotionalCode Component', () => {
    const item = {
        'code': 'ETE2020',
        'description': 'Bénéficier de 20% sur vos achats cet été',
        'endDate': '2021-05-30T00:00:00+00:00',
        'id': 1,
        'label': 'Promotion été 2020',
        'startDate': '2021-04-24T00:00:00+00:00',
    };
    const navigation = {
        'addListener': function addListener(){},
        'canGoBack': function canGoBack(){},
        'closeDrawer': function anonymous(){},
        'dangerouslyGetParent': function dangerouslyGetParent(){},
        'dangerouslyGetState': function anonymous(){},
        'dispatch': function dispatch(){},
        'goBack': function anonymous(){},
        'isFocused': function isFocused(){},
        'jumpTo': function anonymous(){},
        'navigate': function anonymous(){},
        'openDrawer': function anonymous(){},
        'removeListener': function removeListener(){},
        'reset': function anonymous(){},
        'setOptions': function setOptions(){},
        'setParams': function anonymous(){},
        'toggleDrawer': function anonymous(){},
    };
    const myCodes = [
        {
            'code': 'ETE2020',
            'description': 'Bénéficier de 20% sur vos achats cet été',
            'endDate': '2021-05-30T00:00:00+00:00',
            'id': 1,
            'label': 'Promotion été 2020',
            'startDate': '2021-04-24T00:00:00+00:00',
        },
    ];

    test ('Component prints the code (like "ETE2020")', async () => {
        const component = (
            <NavigationContainer>
                <PromotionalCode code={ item } navigation={ navigation } myCodes={ myCodes } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const screenPrintsTheCode = await findByText(item.code);

        expect(screenPrintsTheCode).toBeTruthy();
    });

    test ('Component prints the label', async () => {
        const component = (
            <NavigationContainer>
                <PromotionalCode code={ item } navigation={ navigation } myCodes={ myCodes } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const screenPrintsTheLabel = await findByText(myCodes[0].label);

        expect(screenPrintsTheLabel).toBeTruthy();
    });

    // TODO : regarder si je peux savoir s'il y a bien les boutons "infos" et "trash"
});
