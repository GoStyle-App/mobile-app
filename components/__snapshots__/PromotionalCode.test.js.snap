// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PromotionalCode Component Correct render of component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 10,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#000",
        "color": "#ffffff",
        "padding": 2,
        "textAlign": "center",
        "width": 80,
      }
    }
  >
    ETE2020
  </Text>
  <Text
    style={
      Object {
        "marginLeft": 10,
        "textAlign": "left",
        "width": 150,
      }
    }
  >
    Promotion été 2020
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 15,
            "height": 30,
            "width": 30,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    <View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "#64b5f6",
              "fontSize": 20,
            },
            Array [
              Object {
                "lineHeight": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰋼
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 15,
            "height": 30,
            "width": 30,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    <View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "#f44336",
              "fontSize": 20,
            },
            Array [
              Object {
                "lineHeight": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰩺
      </Text>
    </View>
  </View>
</View>
`;
